torch/distributed/_composable/replicate.py

```diff
@@ -22,6 +18,9 @@ def mark_modules(self, *modules: nn.Module) -> None:
             self.modules.append(module)
             replicate.state(module)._distributed_state = self
             replicate.state(module)._params_collected = False
+            module.register_forward_pre_hook(self.forward_pre_hook)
```

Comment:
this problem to support multiple modules is here, it registered hooks per module and called init_helper() per module as well, but init_helper() will only collect the parameters for the first module, not rest of them, so it is not actually collecting parameters for all the self.modules, right?